google_api_lib/tasks.py

```diff
@@ -461,21 +462,48 @@ def extract_id_from_drive_item_name(drive_item_name) -> Optional[str]:
     return result.group(1) if result else None
 
 
-def get_puzzler_from_uid(uid) -> Optional[str]:
+def get_puzzler_from_uid(google_api_client, uid) -> Optional[str]:
     UserModel = get_user_model()
     try:
         return UserModel.objects.get(social_auth__uid=uid)
     except UserModel.DoesNotExist:
+        pass
+
+    # Fall back to People API
+    # Note that this API depends on the email being added as service account's contact list
+    # It also has a tight request limit
+    time.sleep(2)
```

Comment:
Why do we need to sleep here?

Reply:
So this branch of the code should only run for backward compatibility with previous year's data. When that happens, we actually call the People API in quick succession and it's easy to hit the API's 90 request / minute quota, probably could do with just 1 second though.

Reply:
Oh gotcha. Could you mention that the sleep is to prevent exceeding Google API limits?

Reply:
I guess you mentioned the tight request limit, nvm